ci/stack-orchestration/manual-stacks/txma/template.yml

```diff
@@ -0,0 +1,201 @@
+      production: 'arn:aws:iam::451773080033:root'
+
+Conditions:
+  TxMAIntegrationEnabled:
+    Fn::Or:
+      - Fn::Equals: [!Ref Environment, "staging"]
+      - Fn::Equals: [!Ref Environment, "integration"]
+      - Fn::Equals: [!Ref Environment, "production"]
+
+Resources:
+  TxMASQSProducerAuditEventQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 1209600 # 14 days in seconds
+      QueueName: !Sub '${AWS::StackName}-AuditEventQueue'
+      KmsMasterKeyId: !If
+          - TxMAIntegrationEnabled
+          - !Ref TxMASQSProducerAuditEventQueueEncryptionKeyAlias
+          - !Ref AWS::NoValue
+      RedriveAllowPolicy:
+        redrivePermission: denyAll
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt TxMAAuditEventDeadLetterQueue.Arn
+        maxReceiveCount: 10
+      Tags:
+        - Key: 'AuthoredBy'
+          Value: 'TxMA'
+        - Key: 'ProducerType'
+          Value: 'SQS'
+
+  TxMASQSProducerAuditEventQueuePolicy:
+    Condition: TxMAIntegrationEnabled
+    Type: AWS::SQS::QueuePolicy
+    Properties:
+      Queues:
+        - !Ref TxMASQSProducerAuditEventQueue
+      PolicyDocument:
+        Statement:
+          - Sid: 'AllowReadByTxMAAccount'
+            Effect: Allow
+            Principal:
+              AWS: !FindInMap [TxMAAccountARN, Environment, !Ref 'Environment']
+            Action:
+              - 'sqs:ChangeMessageVisibility'
+              - 'sqs:ReceiveMessage'
+              - 'sqs:DeleteMessage'
+              - 'sqs:GetQueueAttributes'
+            Resource: !GetAtt TxMASQSProducerAuditEventQueue.Arn
+
+  TxMAAuditEventDeadLetterQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 1209600 # 14 days in seconds
+      QueueName: !Sub '${AWS::StackName}-AuditEventDeadLetterQueue'
+      KmsMasterKeyId: !If
+          - TxMAIntegrationEnabled
+          - !Ref TxMASQSProducerAuditEventQueueEncryptionKeyAlias
+          - !Ref AWS::NoValue
+      Tags:
+        - Key: 'AuthoredBy'
+          Value: 'TxMA'
+        - Key: 'ProducerType'
+          Value: 'SQS'
+
+  TxMASQSProducerAuditEventQueueEncryptionKey:
+    Condition: TxMAIntegrationEnabled
+    Type: AWS::KMS::Key
+    Properties:
+      Description: Symmetric key used to encrypt TxMA audit messages at rest in SQS
+      EnableKeyRotation: true
+      KeySpec: SYMMETRIC_DEFAULT
+      KeyPolicy:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: 'Enable root account access'
+            Effect: Allow
+            Principal:
+              AWS: !Sub 'arn:aws:iam::${AWS::AccountId}:root'
+            Action:
+              - 'kms:*'
+            Resource: '*'
+          - Sid: 'Allow Lambda Access'
+            Effect: Allow
+            Principal:
+              Service: 'lambda.amazonaws.com'
+            Action:
+              - kms:Encrypt
+              - kms:Decrypt
+              - kms:ReEncrypt
+              - kms:GenerateDataKey
+            Resource: '*'
```

Comment:
What is this statement for?

Reply:
Nothing, right? 
The previous statement allows the root account to perform KMS actions on all KMS resources. This statement allows all lambdas to perform the listed actions on all KMS resources. Therefore this statement is redundant (and potentially insecure as it allows all lambdas across AWS, not just those in the root account).

Reply:
Good spot, glad that this got picked up.  Well done.